iroha/src/block_sync.rs

```diff
@@ -52,98 +74,131 @@ impl BlockSynchronizer {
             gossip_period: Duration::from_millis(config.gossip_period_ms),
             batch_size: config.batch_size,
             n_topology_shifts_before_reshuffle,
+            broker,
         }
     }
+}
 
-    /// Starts the `BlockSync`, meaning that every `gossip_period`
-    /// the peers would gossip about latest block hashes
-    /// and try to synchronize their blocks.
-    #[log]
-    pub fn start(&self) {
-        let gossip_period = self.gossip_period;
-        let wsv = Arc::clone(&self.wsv);
-        let peer_id = self.peer_id.clone();
-        let sumeragi = Arc::clone(&self.sumeragi);
-        drop(task::spawn(
-            async move {
-                loop {
-                    time::sleep(gossip_period).await;
-                    let message = Message::LatestBlock(LatestBlock::new(
-                        wsv.latest_block_hash(),
-                        peer_id.clone(),
-                    ));
-                    drop(
-                        futures::future::join_all(
-                            sumeragi
-                                .read()
-                                .await
-                                .network_topology
-                                .sorted_peers()
-                                .iter()
-                                .map(|peer| message.clone().send_to(peer)),
-                        )
-                        .await,
-                    );
-                }
-            }
-            .in_current_span(),
+/// Message to send to block synchronizer. It will call `continue_sync` method on it
+#[derive(Debug, Clone, Copy, iroha_actor::Message)]
+pub struct ContinueSync;
+
+/// Message for getting updates from other peers
+///
+/// Starts the `BlockSync`, meaning that every `gossip_period`
+/// the peers would gossip about latest block hashes
+/// and try to synchronize their blocks.
+#[derive(Debug, Clone, Copy, Default, iroha_actor::Message)]
+pub struct RecieveUpdates;
+
+#[async_trait::async_trait]
+impl<S: SumeragiTrait, W: WorldTrait> Actor for BlockSynchronizer<S, W> {
+    async fn on_start(&mut self, ctx: &mut Context<Self>) {
+        self.broker.subscribe::<Message, _>(ctx);
+        self.broker.subscribe::<ContinueSync, _>(ctx);
+        ctx.notify_every::<RecieveUpdates>(self.gossip_period);
+    }
+}
+
+#[async_trait::async_trait]
+impl<S: SumeragiTrait, W: WorldTrait> SimpleHandler<RecieveUpdates> for BlockSynchronizer<S, W> {
+    type Result = ();
+    async fn handle(&mut self, RecieveUpdates: RecieveUpdates) {
```

Comment:
We might add arguments in future. So if we write in this manner, then it will fail to compile on change. Also it is like discarding of message:
```
async fn handle(&mut self, _: RecieveUpdates) {
    ...
}
```

But it is better because it would require changing if structure will be changed.